uitools/import/Esri/ArcGISRuntime/Toolkit/Controller/LocatorSearchSource.qml

```diff
@@ -0,0 +1,390 @@
+        property string searchText: ""
+        onSearchTextChanged: {
+            if (locatorSearchSource.locatorTask && locatorSearchSource.locatorTask.suggestions) {
+                locatorSearchSource.locatorTask.suggestions.searchText = searchText;
+            }
+        }
+
+        signal suggestionsAdded(var source, int first, int last)
+        onRowsInserted: suggestionsAdded(this, first, last)
+
+        signal suggestionsRemoved(var source, int first, int last)
+        onRowsAboutToBeRemoved: suggestionsRemoved(this, first, last)
+    }
+
+    /*!
+      \qmlsignal LocatorSearchSource::searchCompleted(list<ListElement> searchResults)
+      Signal emitted when the search completes, giving the found search results as \a searchResults.
+
+      A result is a ListElement with the following fields.
+
+      \table
+      \header
+         \li name
+         \li type
+         \li description
+      \row
+        \li \c displayTitle
+        \li \c string
+        \li The result heading.
+      \row
+        \li \c displaySubtitle
+        \li \c string
+        \li Optional result subheading.
+      \row
+        \li \c markerImageUrl
+        \li \c url
+        \li URL of the image to display alongside result.
+      \row
+        \li \c owningSource
+        \li \c object
+        \li The search-source this result was created from.
+      \row
+        \li \c geoElement
+        \li \c GeoElement
+        \li The optional map location/graphic of this result.
+      \row
+        \li \c selectionViewpoint
+        \li \c Envelope
+        \li An area containing the result.
+      \endtable
+     */
+    signal searchCompleted(var searchResults)
+
+    /*!
+      \qmlmethod void LocatorSearchSource::search(var query, Geometry area)
+      \brief Starts a search using \a query as input.
+
+      \a query can be a \c string or a \c SuggestResult.
+
+      If \a area is non-null, search is restricted to that area.
+     */
+    function search(query, area) {
+        if (!locatorTask) {
+            searchCompleted([]);
+            return;
+        }
+        if (typeof query === "string") {
+            internal.search_with_method(locatorTask.geocodeWithParameters, query, area);
+        } else if (typeof query === "object" && query instanceof SuggestResult) {
+            internal.search_with_method(locatorTask.geocodeWithSuggestResultAndParameters, query, area);
+        } else {
+            throw `Bad query: ${query}`;
+        }
+    }
+
+    /*!
+        \internal
+     */
+    property QtObject internal: QtObject {
+        id: internal
+
+        property string currentSearchId: ""
+
+        // We're battling the garbage collector here - it's buggy.
+        // This object keeps references to the generated results and will be replaced
+        // on each new subsequent search. This allows the Graphic and Extent of the results
+        // to exist for the appropriate amount of time.
+        property var tempParent: null
```

Comment:
I think it may be a QML issue. The contents of `ListModel` don't seem to be kept alive. I'm not sure if this is a general problem with `ListModel` itself, or if this is a symptom of a more general problem.

Reply:
Thanks for the info. Unfortunately I think that is just how the models access the role data from the object. From my previous experience, our API objects get accessed in a subtle, behind-the-scenes way and it leads to strange situations like what you are seeing.